sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala

```diff
@@ -269,10 +274,6 @@ object DateTimeUtils {
           i += 3
         } else if (i < 2) {
           if (b == '-') {
-            if (i == 0 && j != 4) {
-              // year should have exact four digits
```

Comment:
Again, shall we check the number of year digits? what if overflow happens? like `999999999999999999-12-12`

Reply:
sorry, I was thinking too much about the overflow of the whole date and missed the overflow of the year segment.